build_docs

```diff
@@ -434,14 +434,20 @@ if __name__ == '__main__':
     try:
         logging.basicConfig(level=logging.INFO)
         build_docker_image()
-        if ['--self-test'] == argv[1:]:
+        if len(argv) >= 2 and '--self-test' == argv[1]:
```

Comment:
Doesn't this mean `--self-test` must be positionally the first command-line argument then?

Adding `argparse` adds like maybe 4-5 lines of code, with the benefits of both future-proofing and extensibility.

Reply:
It does indeed need to be first, yeah.

I'll look into argparse!

Reply:
So I had looked into argparse when I first implemented this file and I didn't like it then because there are arguments that have to be passed unchanged into the perl script. I *think* that is still the right call for that branch. The `--self-test` stuff feels pretty similar in that my goal is to pass arguments unchanged to make. It feels a little cleaner to force `--self-test` to be the first argument so it is a little more clear what goes to make.